### DISCUSSION TITLE
Task 1: Cross-Sections Questions

### DISCUSSION THREAD

**Discussion by eitan-weinstein** [fusion-energy/neutronics-workshop#214] [Q&A] [answered]
Hello,

I am new to OpenMC and am using this tutorial to familiarize myself with the software for fusion work. I was trying to go through the first Jupyter notebook and could not get the syntax for openmc.plotter.plot_xs to work as far as "this", and "type" are concerned, because the formatting in the notebook as is leads to an error.

Furthermore, I noticed that this notebook was different than what was being demonstrated in the accompanying videos, so I tried to copy the cells from the video into my notebook to see if I would be able to execute those, but that relied on a module called "plotting_utils" that does not seem to exist anywhere that I am able to find. 

Any suggestions for either or both of these issues?

**Comment by shimwell** [comment]
Sorry to hear that,

I think the video is out of date so please ignore that.

The jupyter notebooks do need the latest development branch of openmc to run.

This will be release as veresion 0.13.4 at some point but for now one needs to use the development branch

The docker images that the workshop run on make use of the openmc development branch.

Are you able to let me know if you are running the workshop through the docker image provided?

**Reply by eitan-weinstein** [reply]
I'm not sure. I downloaded OpenMC through the "2.1. Installing on Linux/Mac with Mamba and conda-forge" route on the Installation and Configuration page of OpenMC. So perhaps the answer is no because I didn't use Docker to install OpenMC but rather mamba. Should I retry with Docker?

**Comment by shimwell** [comment]
The latest conda release of openmc is version 0.13.3 so these plotting notebooks and some of the others are not going to work.

I would recommend the docker image yes. It also contains a few other things needed for the workshop (like DAGMC)

If you do want to try installing openmc develop branch here are the instructions

https://docs.openmc.org/en/stable/quickinstall.html#installing-from-source-on-linux-or-mac-os-x

You would just need to add ```git checkout develop``` after the ```cd openmc``` command

Good luck and let me know if this solves it for you.

Also sorry for making the workshop need the develop branch it just had some really nice features that I couldn't resist. Hopefully there will be an openmc release soon and the workshop will go back to using a stable released version that can be installed with conda

**Reply by gonuke** [reply]
Doesn't the conda image also include DAGMC?  What if someone doesn't have permission to install/run docker?

**Reply by shimwell** [reply]
Yep you are right Paul the default conda install for openmc includes DAGMC

The Docker image also include DAGMC.

The difference is that the notebooks currently use some openmc code that is not in the conda install, that is why the docker image makes use of the openmc develop branch. I guess there will be another conda release before the next NEA OpenMC workshop in April. So Conda install is not going to work for all these workshop tasks. 

I did have a go at writing some install scripts but I think they are not in a good reliable state. Although here are a couple of scripts that might be helpful for installing the workshop dependences https://github.com/fusion-energy/neutronics-workshop/tree/main/install_scripts

Another option is to clone the repo and run github [codespaces](https://github.com/features/codespaces)

**Reply by eitan-weinstein** [reply]
I finally managed to install the most updated version of OpenMC from source, but that is bringing me back to the original issue with "this" and "types" in plot_xs. Are there modifications to the syntax itself that I should be including to make this work?

**Reply by shimwell** [reply]
Are you able to post a screenshot. Many thanks

**Reply by eitan-weinstein** [reply]
![Capture](https://github.com/fusion-energy/neutronics-workshop/assets/134320171/3f386784-bc0c-482f-8b3c-df4dbbbb7dec)

**Reply by shimwell** [reply]
Can you add this line please 
```python
print(openmc.__version__)
```

**Reply by eitan-weinstein** [reply]
At the end of the cell?

**Reply by shimwell** [reply]
Anywhere after the ```import openmc``` line and before the error

**Reply by eitan-weinstein** [reply]
![Capture1](https://github.com/fusion-energy/neutronics-workshop/assets/134320171/169facd0-f191-4cf2-91e7-f52987961192)

**Reply by eitan-weinstein** [reply]
No effect

**Reply by shimwell** [reply, marked as answer]
I can see from the 2nd screen shot that it now printed out 0.13.3 for the version number of openmc being used. This function needs the version from the openmc develop branch

**Reply by eitan-weinstein** [reply]
Alright. Thank you. I'll reinstall with develop branch tomorrow morning and let you know how I fare. Much appreciated

**Reply by eitan-weinstein** [reply]
Looks like it's in good shape now. Thanks for all the help.
